### ISSUE TITLE
More flexible handling of data only messages

### ISSUE THREAD

**Issue by saghul** [arnesson/cordova-plugin-firebase#116]
Hi there! First and foremost, thanks a lot for making this plugin available, I was up and running in minutes! ❤️ 

There is a usecase which doesn't seem to be properly covered (feel free to correct me if I'm wrong here!) so I'd like to share some ideas / options.

It all started when I wanted to have a "heads-up notification" on Android.  No matter how hard I tried I couldn't get the thing to work, so I decided to test something: using data messages instead of notification messages. I see the plugin converts them into "local notifications" [here](https://github.com/arnesson/cordova-plugin-firebase/blob/master/src/android/FirebasePluginMessagingService.java#L81-L107), so with a one liner one can enable that:

~~~
.setPriority(NotificationCompat.PRIORITY_HIGH)
~~~

Back to iOS, looks like data only messages [are not handled at all](https://github.com/arnesson/cordova-plugin-firebase/blob/master/src/ios/AppDelegate%2BFirebasePlugin.m#L124-L127), oops!

So, in light of this, my humble proposal is as follows: drop the local notification on Android and fire a new callback on both platforms ("onDataMessage" perhaps?) then the user can decide what to do.  Maybe they don't want to post a notification, maybe they do!  If they do, they can use [this plugin](https://github.com/katzer/cordova-plugin-local-notifications) which should do the trick.

Thoughts?  If we can agree on the path forward I might be able to provide a patch.

**Comment by robertarnesson** [comment]
Sounds good to me

**Comment by saghul** [comment]
Excellent, I'll cook up a patch then!

**Comment by saghul** [comment]
Well, thatw as too good to be true.  I started implementing the Android part but I ran into a showstopper: there is no way to call any JavaScript callbacks when the app is in the background.  I don't know why I didn't remember this earlier :-S

Given this limitation, the only thing that can be done is fire a notification for the user to click, which brings back the main app activity to the foreground and allows for JavaScript execution. Oh well.

I'm going to close this and open a new issue for the heads-up notifications part.
